### ISSUE TITLE
Upgrading to 4.9.0 adds hostpath mount for falco-audit

### ISSUE THREAD

**Issue by jemag** [falcosecurity/charts#754]
**Describe the bug**

Upgrading to 4.9.0 adds hostpath mount for falco-audit: `proc-fs` mounting hostPath `/proc`

**How to reproduce it**

Install new 4.9.0 chart, can use the example values-k8saudit.yaml

**Expected behaviour**

Falco-audit should not need hostpath to work

**Comment by jemag** [comment]
I am guessing it should probably be conditional to driver being enabled as other hostPath

**Comment by alacuku** [comment]
Hey @jemag, `/proc` is needed for the plugins. We can't make it optional, please see: 
* https://github.com/falcosecurity/charts/pull/739/commits/9efbce0deef5dc03916f9d2a98c586e0b9e59e9c

**Comment by jemag** [comment]
Thank you for the clarification. I couldn't think of why the audit plugin might need it, but I guess it is to add supplementary information about the host.

Can that be a problem for plugins running as deployment such as the audit plugin? Technically, the replica could run on any host, which may or may not be different from each other.

**Comment by Issif** [comment]
> Thank you for the clarification. I couldn't think of why the audit plugin might need it, but I guess it is to add supplementary information about the host.

exactly

> Can that be a problem for plugins running as deployment such as the audit plugin? Technically, the replica could run on any host, which may or may not be different from each other.

whatever the installation kind (deployment or daemonset), it should not have any impact

**Comment by jemag** [comment]
> whatever the installation kind (deployment or daemonset), it should not have any impact

I just meant that in the daemonset/syscall case, events are being generated on the machine itself so the machine is usually 100% representative and additional information can accurately augment these events. This is not the case with most plugins, especially those in deployment since they might be placed on a variety of different VMs, which means this supplementary information will change based on the replica location.

---

The last thing I was wondering: in my mind, Falco plugins may ingest any arbitrary stream of events, which we can then apply rules on. If we these events truly can be coming from any source with the power of plugins, then for a lot of them, it may not make sense to retrieve this supplementary information.

**If a plugin is created to consume events from a Cloud SaaS service, for example, there wouldn't be much need or relevance to get supplementary information from the VM the plugin is running on. In this case, it seems preferable to avoid `hostPath` if its unnecessary permissions granted, for something that does not need it.**

Basically, what I am saying is: given the nature of plugins and the wide possibility of events stream, perhaps it would make sense to make the `proc-fs` `hostPath` optional, if a plugin has no need for it. 

If you think that does not make sense and the `hostPath` is still relevant in every plugins, feel free to close the issue.

**Comment by alacuku** [comment]
I got your point, but the info is not retrieved directly by the plugin but by the `scap_platform`, hence the need to mount the `proc-fs` for all the plugins. Even if a plugin does not need that info, the `scap_platform` will retrieve it and make it available to plugins that depend on it.

I'll leave the PR that introduced it in the libs repository: https://github.com/falcosecurity/libs/pull/1969/files. If we should make the info available only for plugins that need it then feel free to open an issue in the libs repository!

**Comment by jemag** [comment]
Thank you for the clarification. 

I'll check if this can be made optional on the libs repo. If it can, I'll re-open the issue here to ask to be optional as well in the chart.
